src/components/Elements.tsx

```diff
@@ -211,10 +213,21 @@ export const useElementsContextWithUseCase = (
   return parseElementsContext(ctx, useCaseMessage);
 };
 
+const DUMMY_CART_ELEMENT_CONTEXT: CartElementContextValue = {
+  cart: null,
+  cartState: null,
+  setCart: () => {},
+  setCartState: () => {},
+};
+
 export const useCartElementContextWithUseCase = (
-  useCaseMessage: string
+  useCaseMessage: string,
+  isInCustomCheckout = false
 ): CartElementContextValue => {
   const ctx = React.useContext(CartElementContext);
+  if (isInCustomCheckout) {
```

Comment:
This is allow us to keep the interface of `createElementComponent`. 
Note react doesn't allow to **conditionally** call hook. So we cannot choose to call `useCartElementContextWithUseCase` or not based on some conditions.